### DISCUSSION TITLE
[Revealjs] Define new fragment : fade-semi-out-then-in

### DISCUSSION THREAD

**Discussion by fbob** [quarto-dev/quarto-cli#4794] [Q&A] [answered]
I would like to define and use in quarto a new fragment style "fade-semi-out-then-in" i.e text is at first blurred then appear clearly.

From https://revealjs.com/fragments/ it looks like it's possible to define new fragment style with somme css : 

![image](https://user-images.githubusercontent.com/8211124/224676482-6ecf240a-6690-465c-a010-d5b8c8c78801.png)

So I add this to my css config :

```
.fragment.blur {
    filter: blur(5px);
  }
  .fragment.blur.visible {
    filter: none;
```

And I've tried to use it with either : 

```
::: {.fragment .blur}
Test a new fragment syle
:::
```

or

```
::: {.fragment .custom.blur}
Test a new fragment syle
:::
```

But none of them work.
Any idea ?

Thanks !

**Comment by cderv** [comment]
It seems this feature is there since a long time (https://github.com/hakimel/reveal.js/issues/2927) but documented fairly recently (https://github.com/reveal/revealjs.com/pull/50) and the `custom` class support was indeed last week (https://github.com/hakimel/reveal.js/commit/006b348e6b504f3e31b523e8d1438512cc5ad144 and https://github.com/hakimel/reveal.js/issues/2927#issuecomment-1457466446) with some other tweaks. 

The revealjs version inside Quarto is not up to date to this recent change as document in their doc. You should be able to use it the "old way" as explained in https://github.com/hakimel/reveal.js/issues/2927#issuecomment-874724729

Try this 

````markdown
---
title: "My document"
format: 
  revealjs:
    theme: [default, styles.scss]
---

::: {.fragment .blur}
Test a new fragment syle
:::

````

with `styles.scss` being 
````scss
/*-- scss:defaults --*/

/*-- scss:rules --*/

.reveal .slides section .fragment.blur {

  filter: blur(5px);

  &.visible {
    filter: none;
  }
}

````

We'll [update revealjs](https://github.com/quarto-dev/quarto-cli/issues/4795) in 1.4 probably and be sure to add a document this new feature

**Reply by cderv** [reply, marked as answer]
Quarto 1.6 pre-release is now updated with latest revealjs, and custom fragments can be used. 

https://prerelease.quarto.org/docs/presentations/revealjs/advanced.html#custom-fragments

**Comment by fbob** [comment]
Thanks your very much @cderv   for your very helpful and complete answer :+1: 

I've tried your suggestions: it's way better but not quite yet the desired behavior.
The text is at first invisible then blurred.

It's great to see that the revealjs version will be upgraded, hopefully it will work as intended.
Many thanks for your kind and helpful answer !
